pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -1081,3 +1081,16 @@ func StartStaleKeysReclaimer(ctx context.Context) {
 		}
 	}()
 }
+
+// We need to install xfrm state for the local router (cilium_host) early
+// in daemon init path. This is to ensure that we have the xfrm state in
+// place before we advertise the routerIP where other nodes may potentially
+// pick it up and start sending traffic to us. This was previously racing
+// and creating XfrmInNoState errors because other nodes picked up node
+// update before Xfrm config logic was in place. So special case init the
+// rule we need early in init flow.
+func Init() {
+	outerLocalIP := node.GetInternalIPv4Router()
+	wildcardIP := net.ParseIP("0.0.0.0")
+	ipSecReplaceStateIn(outerLocalIP, wildcardIP, false)
```

Comment:
This seems fine iff this will only _add_ state for the target IP and not _replace_, which I guess is implied by the target use case here where the target IP will be different? If for some reason the keys on the filesystem change while Cilium is down and this IP is *not* new, then I suspect this may replace existing state and possibly break existing connections...?

Reply:
It can replace but only if the tuple matches, but tuple in xfrm context also includes SPI so if the cilium_host IP didn't change and keys rolled while we are off restarting we still wont replace because a key roll increments SPI.